### PR TITLE
Default bootstrap servers config to the default if `falsy value`  is provided

### PR DESCRIPTION
**Pull request by kimbugp**
## What does this PR do?

Fixes KafkaClient failing when a `falsy` value is passed in the config for the bootstrap server.
This will default the bootstrap server value to the value from the default config dictionary when a `falsy` value is passed

Passing a `falsy` value return the error 
```
producer = KafkaProducer(bootstrap_servers=None)
File "/opt/ns/nsenv/lib/python2.7/site-packages/kafka/consumer/group.py", line 355, in __init__
    self._client = KafkaClient(metrics=self._metrics, **self.config)
  File "/opt/ns/nsenv/lib/python2.7/site-packages/kafka/client_async.py", line 205, in __init__
    self.cluster = ClusterMetadata(**self.config)
  File "/opt/ns/nsenv/lib/python2.7/site-packages/kafka/cluster.py", line 67, in __init__
    self._bootstrap_brokers = self._generate_bootstrap_brokers()
  File "/opt/ns/nsenv/lib/python2.7/site-packages/kafka/cluster.py", line 72, in _generate_bootstrap_brokers
    bootstrap_hosts = collect_hosts(self.config['bootstrap_servers'])
  File "/opt/ns/nsenv/lib/python2.7/site-packages/kafka/conn.py", line 1485, in collect_hosts
    for host_port in hosts:
TypeError: 'NoneType' object is not iterable
Exception ignored in: <function KafkaClient.__del__ at 0x104330b70>
Traceback (most recent call last):
  File "/opt/ns/nsenv/lib/python2.7/site-packages/kafka/client_async.py", line 441, in __del__
    self._close()
  File "/opt/ns/nsenv/lib/python2.7/site-packages/kafka/client_async.py", line 415, in _close
    if not self._closed:
AttributeError: 'KafkaClient' object has no attribute '_closed'
```
Found that this is because collect_hosts function doesn't check when the host list is empty so the iterator fails with a `TypeError`

---
This change is [Reviewable](https://reviewable.io/reviews/dpkp/kafka-python/1991)

### CODE REVIEW

**Comment by dpkp** [comment]
Can you explain what configuration value you want to support here? Why would someone pass `bootstrap_servers=False` ? Right now I think the best solution is to fix the application to not pass that as configuration. So I'm skeptical about adding one-off fixes into kafka-python to allow bad configuration data.

**Comment by kimbugp** [comment]
@dpkp  In most cases, the bootstrap_servers are got as environment variables. Hence the configuration can be set to None or any other `falsy` value causing the error am trying to fix for.
I am suggesting we support `string` values strictly for the configuration `bootstrap_servers`. We should enforce this even when an iterable is passed as the value  bootstrap_servers.

**Comment by kimbugp** [comment]
> I'm slightly confused here... are there two bug fixes actually as part of this?
> 
> 1. default `falsey` to `localhost`?
> 2. the changes to `conn.py`... looks like you're fixing the `TypeError`, but isn't that a separate error?
> 
> I think both probably need fixing, but it seems they are distinct issues so should probably be separate PR's / commits?
> 
> Thoughts?

@jeffwidman  since the `collect_hosts` functions relates to how the `bootstrap_servers`, are parsed before establishing a connection to the Kafka client, I thought it wise to add this a separate commit in this PR.
But am open to separating it out to another PR if we agree here.

File: kafka/conn.py [hunk 1]
```diff
@@ -1481,9 +1481,13 @@ def collect_hosts(hosts, randomize=True):
         hosts = hosts.strip().split(',')
 
     result = []
+    if not hosts:
+        return result
```
[the remark this answers is not in the document]
Comment (kimbugp) [reply kafka/conn.py]: @jeffwidman, I can go on  to implement this on a new PR  after this discussion https://github.com/dpkp/kafka-python/pull/1991#issuecomment-584598561

File: kafka/client_async.py [hunk 2]
```diff
@@ -201,7 +201,11 @@ def __init__(self, **configs):
         for key in self.config:
             if key in configs:
                 self.config[key] = configs[key]
-
+        if not self.config['bootstrap_servers']:
```
[the remark this answers is not in the document]
Comment (kimbugp) [reply kafka/client_async.py]: @jeffwidman, passing values like `''`, empty iterables etc and all other falsy values in python cause the same error.
Maybe @dpkp can comment on this

**Pull request closed without merging**